pulsar-client/src/main/java/org/apache/pulsar/client/impl/BatchMessageContainerImpl.java

```diff
@@ -141,7 +141,7 @@ public boolean add(MessageImpl<?> msg, SendCallback callback) {
         return isBatchFull();
     }
 
-    protected ByteBuf getCompressedBatchMetadataAndPayload() {
+    protected ByteBuf getCompressedBatchMetadataAndPayload(boolean isBrokerTwoPhaseCompactor) {
```

Comment:
Leaking implementation details that compaction doesn't compress messages is not very great. it would be better to rename `isBrokerTwoPhaseCompactor` to `allowCompression` and inverse the value. Add an override that passes `true` as the default value for `allowCompression`.

Reply:
done